hoodie-client/src/main/java/com/uber/hoodie/io/HoodieAppendHandle.java

```diff
@@ -87,43 +101,46 @@ public HoodieAppendHandle(HoodieWriteConfig config, String commitTime, HoodieTab
     this.recordItr = recordItr;
   }
 
-  private void init(String partitionPath) {
-
-    // extract some information from the first record
-    FileSlice fileSlice = fileSystemView.getLatestFileSlices(partitionPath)
-        .filter(fileSlice1 -> fileSlice1.getDataFile().get().getFileId().equals(fileId)).findFirst()
-        .get();
-    // HACK(vc) This also assumes a base file. It will break, if appending without one.
-    String latestValidFilePath = fileSlice.getDataFile().get().getFileName();
-    String baseCommitTime = FSUtils.getCommitTime(latestValidFilePath);
-    writeStatus.getStat().setPrevCommit(baseCommitTime);
-    writeStatus.setFileId(fileId);
-    writeStatus.setPartitionPath(partitionPath);
-    writeStatus.getStat().setFileId(fileId);
-    this.partitionPath = partitionPath;
+  public HoodieAppendHandle(HoodieWriteConfig config, String commitTime, HoodieTable<T> hoodieTable) {
+    this(config, commitTime, hoodieTable, UUID.randomUUID().toString(), null);
```

Comment:
Not sure if I understand this clearly : Don't we need to pass the correct fileId for AppendHandle for MergeOnRead insert-to-log case ? Otherwise the storage writer initialized in the init() method would be using a fake fileId. right ?

Reply:
Cleared with Nishith. This is used in MergeOnWrite case where a new file-group is getting created.